Add an example deployment yaml file

elezar opened the pull request:
I have added an example deployment file. Note that the Pierone registry and mint bucket need to be specified. Still to be done:

- [x] Document the options in the YAML file (e.g. spot instances)
- [x] Describe the docker build step in the README.
- [x] Describe the stack creation step in the README.

Commit:
Add an example deployment yaml file

Commit:
Add an example deployment yaml file

Commit:
Add section on deployment to the readme

Commit:
Merge branch 'add_senza_file' of github.com:elezar/gpu-hello-world into add_senza_file

Commit:
Add README instructions

Commit:
Specify security group and IAM role in senza file

Commit:
Update README

Commit:
Update readme to match STUPS docs

Commit:
Remove extra region flag

elezar commented:
@mikkeloscar @georghildebrand  I have updated the example in this repo, and will also update the documentation in zalando-stups/documentation#117.

georghildebrand commented:
@elezar , thanks for all your efforts! Looks good to me.

mikkeloscar commented on README.md:
````diff
@@ -1,5 +1,49 @@
+
+The built image can then be pushed to the PierOne registry:
+```bash
+docker push pierone.stups.example.org/<your-team>teamname/gpu-hello-world:IMAGEVERSION
````
teamname is defined twice here.

mikkeloscar commented on deploy-definition.yaml:
```diff
@@ -0,0 +1,74 @@
+      TaupageConfig:
+        application_version: "{{Arguments.ImageVersion}}"
+        runtime: Docker
+        source: "{{Arguments.PieroneRegistry}}/{{AccountInfo.TeamID}}/gpu-hello-world:{{Arguments.ImageVersion}}"
```
Maybe just use the whole image as argument?

```
source: "{{Arguments.DockerImage}}:{{Arguments.ImageVersion}}"
```

mikkeloscar commented on README.md:
````diff
@@ -1,5 +1,49 @@
+In order to deploy the stack with version `STACKVERSION`, run the following `senza` command:
+
+```bash
+senza create --region=eu-west-1 deploy-definition.yaml STACKVERSION pierone.stups.example.org IMAGEVERSION example-mint-bucket-eu-west-1
````
Teamname is missing here, but I think it makes sense to parse the whole `DockerImage` as an argument. (See comment below).

mikkeloscar commented on deploy-definition.yaml:
```diff
@@ -0,0 +1,74 @@
+          ToPort: 22
+          CidrIp: 0.0.0.0/0
+
+  GPUHelloWorldRole:
```
Is this needed? Taupage no longer need it to pull images from pierone, so it could be left out. Maybe you put it here to not confuse people if they need credentials for pulling data? You know the audience better than me :)

elezar replied:
I was just trying to follow the standard examples. It could be that this is not needed. How would the alternative look?

mikkeloscar replied:
You would just leave the whole role definition out (and the role reference above).

elezar replied:
But this would require that the role be required manually, e.g. to access the mint bucket. In this way there is no need to first run a "dummy" `senza init`.

mikkeloscar replied:
My point was that, if the application you are running doesn't need oauth credentials because it doesn't need to connect to other services, then the role is not needed at all.

The fact that we are now discussing this, makes me think it's better to just keep it as is, to not confuse people :)

Commit:
Update README.md

Remove duplicate teamname.

Commit:
Update README.md

Specify `IMAGENAME` in full.

Commit:
Update README.md

Don't use `IMAGENAME`, specify full path.

Commit:
Update deploy-definition.yaml

Use `DockerImage` as a single unit.

elezar commented:
@mikkeloscar I have addressed your comments. Not sure what to do about the Pierone issue here. Do you mean that we can just push to dockerHub or something?

mikkeloscar commented:
:+1:

elezar commented:
:+1:
